include/superior_mysqlpp/dnsa_connection_pool.hpp

```diff
@@ -101,88 +125,106 @@ namespace SuperiorMySqlpp
-                            {
-                                getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementChangeDetected(getBase().getId());
 
-                                getBase().clearPool();
-                                lastIpAddresses = std::move(ipAddresses);
-                            }
-                        }
-                        /*
-                         * DNS is sometimes to fail...
-                         */
-                        catch (std::exception& e)
-                        {
-                            getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementCheckError(getBase().getId(), e);
-                        }
-                        catch (...)
+                        // Check if ip addresses match to last resolved ip addresses.
+                        if (!std::equal(lastIpAddresses.begin(), lastIpAddresses.end(), ipAddresses.begin(), ipAddresses.end()))
                         {
-                            getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementCheckError(getBase().getId());
-                        }
-
-                        getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementCycleEnd(getBase().getId());
+                            getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementChangeDetected(getBase().getId());
 
-                        sleepInParts(sleepTime, std::chrono::milliseconds{50}, [&](){ return enabled.load(); });
+                            getBase().clearPool();
+                            lastIpAddresses = std::move(ipAddresses);
+                        }
+                    }
+                    // catch all dns related errors
+                    catch (std::exception& e)
+                    {
+                        getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementCheckError(getBase().getId(), e);
+                    }
+                    catch (...)
+                    {
+                        getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementCheckError(getBase().getId());
                     }
 
-                    getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementStopped(getBase().getId());
-                }
-                catch (std::exception& e)
-                {
-                    getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementError(getBase().getId(), e);
-                    onError();
-                }
-                catch (...)
-                {
-                    getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementError(getBase().getId());
-                    onError();
+                    getBase().getLogger()->logSharedPtrPoolDnsAwarePoolManagementCycleEnd(getBase().getId());
+
+                    sleepInParts(sleepTime, std::chrono::milliseconds{50}, [&](){ return enabled.load(); });
```

Comment:
I think that 50ms should probably be at least some kind of "explicit" constant?

Reply:
What about "shutdownCheckPeriod"?